server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimiter.scala

```diff
@@ -67,6 +68,31 @@ class RedisRateLimiterFactory @Inject()(redisConfiguration: RedisConfiguration)
       masterReplicaRedisConfiguration.redisURI.value.asJava,
       masterReplicaRedisConfiguration.readFrom)
 
+    case sentinelRedisConfiguration: SentinelRedisConfiguration => new RedisSingleInstanceRateLimitjFactory(RedisClient.create(sentinelRedisConfiguration.redisURI))
+
+    case tlsStandaloneRedisConfiguration: TLSStandaloneRedisConfiguration =>
```

Comment:
The complexity of injecting the redis client is leaking into the call sites.


Is this intended?

Shouldn't this place just be injected the RedisClient? Why is it instanciating it itself?